From 541885c6035266903c9f9057ff070b9f67edc8b3 Mon Sep 17 00:00:00 2001
From: Sam James <sam@gentoo.org>
Date: Wed, 2 Aug 2023 12:33:19 +0100
Subject: [PATCH] Switch to libbsd-overlay

This is more portable than libbsd as everything Just Works, even on BSD systems,
and is the recommended method of consuming libbsd nowadays.

It also helpfully lets things work with glibc-provided functions for new
enough glibc.

Closes: https://gitlab.freedesktop.org/xorg/xserver/-/merge_requests/973
Co-authored-by: Guillem Jover <guillem@hadrons.org>
Signed-off-by: Sam James <sam@gentoo.org>
---
 include/os.h | 9 +++------
 meson.build  | 2 +-
 os/auth.c    | 4 +---
 3 files changed, 5 insertions(+), 10 deletions(-)

diff --git a/include/os.h b/include/os.h
index bb3348b185..7db24088f8 100644
--- a/include/os.h
+++ b/include/os.h
@@ -50,16 +50,13 @@ SOFTWARE.
 #include "misc.h"
 #include <stdarg.h>
 #include <stdint.h>
+#if defined(HAVE_REALLOCARRAY)
+#include <stdlib.h>       /* for reallocarray */
+#endif
 #include <string.h>
 #ifdef MONOTONIC_CLOCK
 #include <time.h>
 #endif
-#if defined(HAVE_LIBBSD) && defined(HAVE_REALLOCARRAY)
-#include <bsd/stdlib.h>       /* for reallocarray */
-#endif
-#if defined(HAVE_LIBBSD) && defined(HAVE_STRLCPY)
-#include <bsd/string.h>       /* for strlcpy, strlcat */
-#endif
 
 #define SCREEN_SAVER_ON   0
 #define SCREEN_SAVER_OFF  1
diff --git a/meson.build b/meson.build
index 515cc44eaf..0279ea211f 100644
--- a/meson.build
+++ b/meson.build
@@ -98,7 +98,7 @@ xshmfence_dep = dependency('xshmfence', version: '>= 1.1', required: false)
 xwaylandproto_dep = dependency('xwaylandproto', version: '>= 1.0', fallback: ['xorgproto', 'ext_xorgproto'], required: false)
 
 pixman_dep = dependency('pixman-1')
-libbsd_dep = dependency('libbsd', required: false)
+libbsd_dep = dependency('libbsd-overlay', required: false)
 xkbcomp_dep = dependency('xkbcomp', required: false)
 xkbfile_dep = dependency('xkbfile')
 xfont2_dep = dependency('xfont2', version: '>= 2.0')
diff --git a/os/auth.c b/os/auth.c
index 611e4efb1c..243d3c5a8c 100644
--- a/os/auth.c
+++ b/os/auth.c
@@ -46,9 +46,7 @@ from The Open Group.
 #ifdef WIN32
 #include    <X11/Xw32defs.h>
 #endif
-#ifdef HAVE_LIBBSD
-#include   <bsd/stdlib.h>       /* for arc4random_buf() */
-#endif
+#include   <stdlib.h>       /* for arc4random_buf() */
 
 struct protocol {
     unsigned short name_length;
-- 
GitLab

